Packet++/src/VrrpLayer.cpp

```diff
@@ -0,0 +1,517 @@
+#define LOG_MODULE PacketLogModuleVrrpLayer
+
+#include <SystemUtils.h>
+#include "PacketUtils.h"
+#include "Logger.h"
+#include "EndianPortable.h"
+#include "IPv4Layer.h"
+#include "PayloadLayer.h"
+#include "VrrpLayer.h"
+
+namespace pcpp {
+
+	/*************
+	 * VrrpLayer
+	 *************/
+
+	VrrpLayer::VrrpLayer(ProtocolType subProtocol) {
+		m_DataLen = VRRP_PACKET_FIX_LEN;
+		m_Data = new uint8_t[m_DataLen];
+		memset(m_Data, 0, m_DataLen);
+		m_Protocol = subProtocol;
+		m_AddressType = IPAddress::IPv4AddressType;
+	}
+
+	ProtocolType VrrpLayer::getVersionFromData(uint8_t *data, size_t dataLen) {
+		if (dataLen <= VRRP_PACKET_FIX_LEN) {
+			return UnknownProtocol;
+		}
```

Comment:
This check should be:
```cpp
if (!data || dataLen < sizeof(vrrp_packet))
{
	return UnknownProtocol;
}
```

Reply:
Fixed in https://github.com/seladb/PcapPlusPlus/pull/1118/commits/6bcaa96a4cf7117ccb736e5224f0ea178d6bf1a3